lib/wasi/src/lib.rs

```diff
@@ -111,49 +109,92 @@ pub use crate::{
         WasiEnv, WasiEnvBuilder, WasiEnvInit, WasiFunctionEnv, WasiInstanceHandles,
         WasiStateCreationError, ALL_RIGHTS,
     },
-    syscalls::types,
+    syscalls::{rewind, types, unwind},
     utils::{get_wasi_version, get_wasi_versions, is_wasi_module, WasiVersion},
 };
 
-/// This is returned in `RuntimeError`.
-/// Use `downcast` or `downcast_ref` to retrieve the `ExitCode`.
-#[derive(Error, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub enum WasiError {
-    #[error("WASI exited with code: {0}")]
-    Exit(ExitCode),
-    #[error("The WASI version could not be determined")]
-    UnknownWasiVersion,
+/// Trait that will be invoked after the rewind has finished
+/// It is possible that the process will be terminated rather
+/// than restored at this point
+pub trait RewindPostProcess {
```

Comment:
Maybe put this in `os/task/rewind.rs`?

We could collect related functionality there.

Reply:
yeah might be good to create that but not in `/os` as its more runtime related

Reply:
https://github.com/wasmerio/wasmer/pull/3710/commits/4b7ddf145b70bf832e4ba167b1390c27def0b3f3